### PR TITLE
support for flutter 3.0.0

### PR DESCRIPTION
**Pull request by JacobSamro**

### CODE REVIEW

**Comment by erickok** [comment]
Seems to be blocked mostly due to the flutter_widget_from_html_core dependency.

**Comment by appcapsergen** [comment]
Could it be considered to upgrade the xml dependency regarding the following issue?
https://github.com/jhomlala/betterplayer/issues/1013

File: pubspec.yaml [hunk 1]
```diff
@@ -17,8 +17,8 @@ dependencies:
   cupertino_icons: ^1.0.4
   wakelock: ^0.6.1+2
   meta: ^1.7.0
-  flutter_widget_from_html_core: ^0.8.5+1
-  visibility_detector: ^0.2.2
+  flutter_widget_from_html_core: ^0.8.5+3
+  visibility_detector: ^0.3.3
   path_provider: ^2.0.9
   collection: ^1.15.0
   xml: ^5.3.0
```
Comment (junaid1460) [inline pubspec.yaml:24]: upgrade xml?

Comment (appcapsergen) [reply pubspec.yaml:24]: Yes, the xml dependency on line 24
`xml: ^5.3.0`

Comment (discoveruz) [reply pubspec.yaml:24]: please upgrade xml lastest version

Comment (ciriousjoker) [reply pubspec.yaml:24]: Updating xml should not be necessary for Flutter 3 support, so I'd argue for merging this and updating the other dependencies in a separate PR.

**Review by ciriousjoker — APPROVED** [review]
lgtm

**Comment by BenCherif** [comment]
@ciriousjoker  as you said updating xml should not be necessary for Flutter 3 support ,but I use another package called flutter_svg, which now depends on xml 6.0.1, which makes it incompatible with better_player, since it depends on xml 5.3.0. check #1013

**Comment by ciriousjoker** [comment]
We're also depending on `flutter_svg` and decided to stick with v1.0.3 for the time being. Seems to work just fine with Flutter 3 and better_player v0.0.82. Don't get me wrong, I'm not against updating the XML dependency, just that both should be done in the same PR which would lead to the same version.

What if someone uses Flutter 3 and an old flutter_svg? Ideally, that person would have at least one in between version even if most people will just update.

**Comment by jhomlala** [comment]
Fixed in latest release.

**Pull request closed without merging**